src/Network/Network/PrivateLinkService/PrivateLinkServiceProvider/ProviderConfiguration.cs

```diff
@@ -71,43 +72,55 @@ static ProviderConfiguration()
             RegisterConfiguration("Microsoft.Web/hostingEnvironments", "2020-10-01", true, false);
             RegisterConfiguration("Microsoft.BotService/botServices", "2021-05-01-preview", true, true);
         }
-
-        private static void RegisterConfiguration(string type, string apiVersion, bool hasConnectionsURI = false, bool hasResourceURI = false)
+        /// <summary>
+        /// Register priavte endopoint connection and private link resource configuration
+        /// </summary>
+        /// <param name="type">Resource type</param>
+        /// <param name="apiVersion">Resource api version</param>
+        /// <param name="hasConnectionsURI">True if the private endpoint connection can be list by URL <see cref="GenericProvider.BuildPrivateEndpointConnectionsURL(string, string)"/>, otherwise it can be list by URL <see cref="GenericProvider.BuildPrivateEndpointConnectionsOwnerURL(string, string)"/></param>
+        /// <param name="supportGetPrivateLinkResource">True if the private link resource can be get by Id, otherwise it can be list</param>
+        /// <param name="supportPrivateLinkResource">True if the private link resource be supported, otherwise false</param>
+        private static void RegisterConfiguration(string type, string apiVersion, bool hasConnectionsURI = false, bool supportGetPrivateLinkResource = false, bool supportPrivateLinkResource = true)
         {
             ProviderConfiguration configuration = new ProviderConfiguration
             {
                 Type = type,
                 ApiVersion = apiVersion,
                 HasConnectionsURI = hasConnectionsURI,
-                HasResourceURI = hasResourceURI
+                SupportGetPrivateLinkResource = supportGetPrivateLinkResource,
+                SupportPrivateLinkResource = supportPrivateLinkResource,
```

Comment:
In previous practice, we consider provider supporting list operation by default. Now we are planning to distinguish them by different flag. I'd like to name these flags by `SupportsGetPrivateLinkResource` and `SupportsListPrivateLinkResource` respectively.
```suggestion
                SupportsGetPrivateLinkResource= supportsGetPrivateLinkResource,
                SupportsListPrivateLinkResource= supportsListPrivateLinkResource,
```

Reply:
Done